policy/src/main/java/brooklyn/enricher/RollingTimeWindowMeanEnricher.java

```diff
@@ -103,31 +119,69 @@ public void onEvent(SensorEvent<T> event) {
     public void onEvent(SensorEvent<T> event, long eventTime) {
         values.addLast(event.getValue());
         timestamps.addLast(eventTime);
-        pruneValues(eventTime);
-        entity.setAttribute((AttributeSensor<Double>)target, getAverage(eventTime).value); //TODO this can potentially go stale... maybe we need to timestamp as well?
+        if (eventTime>0) {
+            ConfidenceQualifiedNumber average = getAverage(eventTime, 0);
+
+            if (average.confidence > getConfig(CONFIDENCE_REQUIRED_TO_PUBLISH)) { 
+                // without confidence, we might publish wildly varying estimates,
+                // causing spurious resizes, so allow it to be configured, and
+                // by default require a high value
+
+                // TODO would be nice to include timestamp, etc
+                entity.setAttribute((AttributeSensor<Double>)target, average.value); 
+            }
+        }
     }
     
     public ConfidenceQualifiedNumber getAverage() {
```

Comment:
Nothing seems to call the `getAverage()` method, so the `TIMESTAMP_GRACE_TIME` is never used. Or am I missing something?